### PR TITLE
Bugfix FXIOS-4949 [v106] Tab pickup tile not showing after onboarding

### PR DESCRIPTION
**Pull request by lmarceau**
# [FXIOS-4949](https://mozilla-hub.atlassian.net/browse/FXIOS-4949) https://github.com/mozilla-mobile/firefox-ios/issues/11940
Update sync account variable from jump back in data adaptor when sync account changes

### CODE REVIEW

File: Client/Frontend/Home/JumpBackIn/Data/JumpBackInDataAdaptor.swift [hunk 1]
```diff
@@ -258,7 +263,9 @@ class JumpBackInDataAdaptorImplementation: JumpBackInDataAdaptor, FeatureFlaggab
         var mostRecentTab: (client: RemoteClient, tab: RemoteTab)?
 
         desktopClientAndTabs.forEach { remoteClient in
-            guard let firstClient = remoteClient.tabs.first else { return }
+            guard let firstClient = remoteClient.tabs.first else {
+                return
+            }
```
Comment (electricRGB) [inline Client/Frontend/Home/JumpBackIn/Data/JumpBackInDataAdaptor.swift:268]:
```suggestion
            guard let firstClient = remoteClient.tabs.first else {
                return
            }
```

nit: Why the change here?

**Review by electricRGB — APPROVED** [review]

**Review by thatswinnie — APPROVED** [review]
